src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -49,16 +44,26 @@ function getInitialInputValues(initialSelectedInputAsset: ExtendedAnimatedAssetW
   const initialInputAmount = niceIncrementFormatter({
     incrementDecimalPlaces: initialDecimalPlaces,
     inputAssetBalance: initialBalance,
+    inputAssetNativePrice: initialSelectedInputAsset?.price?.value ?? 0,
     assetBalanceDisplay,
-    inputAssetUsdPrice: initialSelectedInputAsset?.price?.value ?? 0,
     niceIncrement: initialNiceIncrement,
     percentageToSwap: 0.5,
     sliderXPosition: SLIDER_WIDTH / 2,
     stripSeparators: true,
     isStablecoin,
   });
 
-  const initialInputNativeValue = toFixedWorklet(mulWorklet(initialInputAmount, initialSelectedInputAsset?.price?.value ?? 0), 2);
+  const nativeCurrency = store.getState().settings.nativeCurrency;
```

Comment:
any reason we can't just use `convertAmountAndPriceToNativeDisplay` here?

Reply:
we don't want to include the denomination here